pyfr/solvers/baseadvecdiff/system.py

```diff
@@ -15,16 +15,33 @@ def deps(dk, *names): return self._kdeps(k, dk, *names)
         g1 = self.backend.graph()
         g1.add_mpi_reqs(m['scal_fpts_recv'])
 
+        # Perform post-processing of the previous solution stage
+        g1.add_all(k['eles/filter_solution'])
+
         # Interpolate the solution to the flux points
-        g1.add_all(k['eles/disu'])
+        g1.add_all(k['eles/disu'], deps=k['eles/filter_solution'])
```

Comment:
Can this be made element by element?  Or must all of the filtering for all element types finish before we can do the interpolation for any of them?

Reply:
I think so? Assuming that doesn't allow the interface kernels to run before all of the interpolation kernels are finished

Reply:
Yeah, subsequent interface kernels block on all interpolation kernels.  But back to back element kernels should be able to run once their corresponding element type is done.

Reply:
Something like 
```
        for filt, disu in zip(k['eles/entropy_filter'], k['eles/disu']):
            g1.add(filt)
            ddeps = deps(disu, 'eles/entropy_filter')
            g1.add(disu, deps=ddeps)
```

Reply:
Can do an add_all on the filter to avoid the zip.  Then
```
for l in k['eles/disu']:
    g1.add(l, deps=deps(l, 'eles/entropy_filter'))
```